nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/queue/GetAzureQueueStorage.java

```diff
@@ -0,0 +1,208 @@
+
+        final int visibilityTimeoutInSecs = context.getProperty(VISIBILITY_TIMEOUT).asTimePeriod(TimeUnit.SECONDS).intValue();
+        final int batchSize = context.getProperty(BATCH_SIZE).asInteger();
+        final boolean autoDelete = context.getProperty(AUTO_DELETE).asBoolean();
+
+        final Iterable<CloudQueueMessage> retrievedMessagesIterable;
+
+        try {
+            retrievedMessagesIterable = cloudQueue.retrieveMessages(batchSize, visibilityTimeoutInSecs, null, null);
+        } catch (final StorageException e) {
+            getLogger().error("Failed to retrieve messages from the provided Azure Storage Queue due to {}", new Object[] {e});
+            context.yield();
+            return;
+        }
+
+        final List<CloudQueueMessage> cloudQueueMessages = toList(retrievedMessagesIterable);
+
+        for (final CloudQueueMessage message : cloudQueueMessages) {
+            FlowFile flowFile = session.create();
+
+            final Map<String, String> attributes = new HashMap<>();
+
+            attributes.put("azure.queue.uri", cloudQueue.getUri().toString());
+            attributes.put("azure.queue.insertionTime", message.getInsertionTime().toString());
+            attributes.put("azure.queue.expirationTime", message.getExpirationTime().toString());
+            attributes.put("azure.queue.messageId", message.getMessageId());
+            attributes.put("azure.queue.popReceipt", message.getPopReceipt());
+
+            flowFile = session.putAllAttributes(flowFile, attributes);
+            flowFile = session.write(flowFile, out -> {
+                try {
+                    out.write(message.getMessageContentAsByte());
+                } catch (StorageException e) {
+                    getLogger().error("Failed to write the retrieved queue message to FlowFile content due to {}", new Object[] {e});
+                    context.yield();
+                }
+            });
+
+            session.transfer(flowFile, REL_SUCCESS);
+            session.getProvenanceReporter().receive(flowFile, cloudQueue.getStorageUri().toString());
+        }
+
+        if(autoDelete) {
+            session.commit();
+
+            for (final CloudQueueMessage message : cloudQueueMessages) {
+                try {
+                    cloudQueue.deleteMessage(message);
+                } catch (StorageException e) {
+                    getLogger().error("Failed to delete the retrieved message with the id {} from the queue due to {}",
+                            new Object[] {message.getMessageId(), e});
+                }
+            }
+        }
+
+    }
+
+    @Override
+    public Collection<ValidationResult> customValidate(final ValidationContext validationContext) {
+        final List<ValidationResult> problems = new ArrayList<>(super.customValidate(validationContext));
+
+        final int batchSize = validationContext.getProperty(BATCH_SIZE).asInteger();
```

Comment:
I believe you can set a range validator directly in the property (IIRC it's in CommonValidator or something like that)

Reply:
Done. It is `createLongValidator` although I feel it could have been better named in order to be easily found while developing time itself :)